File: c/03-estrutura-repetitiva/soma_impares.c
/*Problema "soma_impares" (adaptado de URI 1071)
Leia 2 valores inteiros X e Y (em qualquer ordem). A seguir, calcule e mostre a soma dos números
impares entre eles.*/


#include <stdio.h> 
int main () {
    int x, y, max, min, sum = 0;

    printf("Digite dois numeros: \n");
    scanf("%d %d", &x, &y);

    if(x > y) {
        max = x;
        min = y;
    } else {
        max = y;
        min = x;
    }

    for(int i = min + 1; i < max; i++) {
        if (i % 2 != 0) {
            sum += i;
        }
    }

    printf("SOMA DOS IMPARES = %d\n", sum);

return 0;
}

File: python/03-estruturas-repetitivas/soma_impares.py
"""Problema "soma_impares" (adaptado de URI 1071)
Leia 2 valores inteiros X e Y (em qualquer ordem). A seguir, calcule e mostre a soma dos números
impares entre eles."""

print("Digite dois numeros:")
x = int(input())
y = int(input())

if x > y:
    maior = x
    menor = y
else:
    maior = y
    menor = x

soma = 0

# O range(inicio, fim) não inclui o valor de 'fim', 
# exatamente como o seu 'i < max' no C++.
for i in range(menor + 1, maior):
    if i % 2 != 0:
        soma += i

print(f"SOMA DOS IMPARES = {soma}")